tests/common/devices/sonic.py

```diff
@@ -530,10 +581,50 @@ def all_critical_process_status(self):
         """
         @summary: Check whether all critical processes status for all critical services
         """
-        result = {}
+        # Get critical process definition of all services
+        group_process_results = self.critical_group_process()
```

Comment:
Can `all_critical_process_status` call `critical_process_status` ?

Reply:
Yes, it can. However, it would be less efficient. The `all_critical_process_status` method runs all the commands to check processes of all critical services in a single batch. Method `critical_process_status` only runs commands to check processes of a single service in a batch. If we reuse `critical_process_status` for `critical_process_status`, multiple batch of commands would be required, one batch for one critical service. To have better efficiency, we may need to tolerate some code duplication.

Reply:
It makes sense. Thanks